components/tools/OmeroWeb/omeroweb/feedback/views.py

```diff
@@ -131,12 +130,13 @@ def send_comment(request):
 ##############################################################################
 # handlers
 
-
 def csrf_failure(request, reason=""):
-    logger.warn('csrf_failure: Forbidden')
-    t = template_loader.get_template("403_csrf.html")
-    c = RequestContext(request, {})
-    return HttpResponseForbidden(t.render(c))
+    """
+    Always return Json response
+    since this is accepted by browser and API users
+    """
+    error = "CSRF Error. You need to include 'X-CSRFToken' in header"
```

Comment:
I think `csrf_failure` is used for any request, not just ajax. The error message is not enough generic. `X-CSRFToken` is only set on on each XMLHttpRequest, otherwise cookie is used.